layout-management-services/src/main/java/org/exoplatform/layoutmanagement/rest/SiteNavigationRestService.java

```diff
@@ -489,14 +503,83 @@ public Response getPages(@Context
+                             @Parameter(description = "page site type", required = true)
+                             @QueryParam("pageSiteType")
+                             String pageSiteType,
+                             @Parameter(description = "page site name", required = true)
+                             @QueryParam("pageSiteName")
+                             String pageSiteName,
+                             @Parameter(description = "Page Type : GROUP, PAGE OR LINK", required = true)
+                             @QueryParam("pageType")
+                             String pageType,
+                             @Parameter(description = "link")
+                             @QueryParam("link")
+                             String link,
+                             @Parameter(description = "page template : blank , normal, analytics ...")
+                             @QueryParam("pageTemplate")
+                             String pageTemplate) {
+    if (StringUtils.isBlank(pageName) || StringUtils.isBlank(pageType) || StringUtils.isBlank(pageSiteName)
+        || StringUtils.isBlank(pageSiteType)) {
+      return Response.status(Response.Status.BAD_REQUEST).entity("params are mandatory").build();
+    }
+    try {
+      Page page;
+      if (PageType.PAGE.equals(PageType.valueOf(pageType))) {
+        if (StringUtils.isBlank(pageTemplate)) {
+          return Response.status(Response.Status.BAD_REQUEST).entity("pageTemplate param is mandatory for PAGE pageType").build();
+        }
+        page = userPortalConfigService.createPageTemplate(pageTemplate, pageSiteType, pageSiteName);
+      } else {
+        page = new Page(pageSiteType, pageSiteName, pageName);
+      }
+      page.setName(pageName);
+      page.setTitle(pageName);
+      page.setType(pageType);
+      setDefaultPermission(page, new SiteKey(pageSiteType, pageSiteName));
+      PageState pageState = Utils.toPageState(page);
+      layoutService.save(new PageContext(page.getPageKey(), pageState), page);
+      PageContext createdPage = layoutService.getPageContext(page.getPageKey());
+      return Response.ok().entity(createdPage).build();
+    } catch (Exception e) {
+      LOG.error("Error when creating a new page", e);
+      return Response.serverError().entity(e.getMessage()).build();
+    }
+  }
+
+  private void setDefaultPermission(Page page, SiteKey siteKey) {
+    if (SiteType.PORTAL.equals(siteKey.getType())) {
+      page.setAccessPermissions(new String[]{ UserACL.EVERYONE });
+      if (page.getEditPermission() == null || page.getEditPermission().isEmpty()) {
+        page.setEditPermission("manager:/platform/administrators");
+      }
+    } else if (SiteType.GROUP.equals(siteKey.getType())) {
+      String siteName = siteKey.getName().startsWith("/") ? siteKey.getName() : "/" + siteKey.getName();
+      page.setAccessPermissions(new String[] { "*:" + siteName });
+      if (page.getEditPermission() == null || page.getEditPermission().isEmpty()) {
```

Comment:
this test is not needed IMO since the page is not yet created, so no edit permission yet